Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -1141,21 +1194,33 @@ internal void MirrorUpdate()
         {
             if (observers != null && observers.Count > 0)
             {
-                NetworkWriter writer = NetworkWriterPool.GetWriter();
+                // one writer for owner, one for observers
+                NetworkWriter ownerWriter = NetworkWriterPool.GetWriter();
+                NetworkWriter observersWriter = NetworkWriterPool.GetWriter();
+
                 // serialize all the dirty components and send (if any were dirty)
-                if (OnSerializeAllSafely(false, writer))
+                if (OnSerializeAllSafely(false, ownerWriter, observersWriter))
                 {
                     // populate cached UpdateVarsMessage and send
                     varsMessage.netId = netId;
-                    // segment to avoid reader allocations.
-                    // (never null because of our above check)
-                    varsMessage.payload = writer.ToArraySegment();
-                    NetworkServer.SendToReady(this, varsMessage);
+
+                    // send ownerWriter to owner
+                    // (only if there is a connection (e.g. if not a monster),
+                    //  and if connection is ready because we use SendToReady
+                    //  below too)
+                    varsMessage.payload = ownerWriter.ToArraySegment();
+                    if (connectionToClient != null && connectionToClient.isReady)
+                        NetworkServer.SendToClientOfPlayer(this, varsMessage);
+
+                    // send observersWriter to everyone but owner
+                    varsMessage.payload = observersWriter.ToArraySegment();
+                    NetworkServer.SendToReady(this, varsMessage, false);
```

Comment:
The whole point of synctoowner is that we don't have to notify observers of changes that are only relevant to the owner.

But in this line,  we send them an empty message anyway even if only owner data changed.

Reply:
nice find @paulpach . fixed it.